google-cloud-clients/google-cloud-bigtable/src/main/java/com/google/cloud/bigtable/admin/v2/TableAdminClient.java

```diff
@@ -0,0 +1,757 @@
+   * bigtableTableAdminSettings
+   *
+   * @param instanceName
+   * @param adminSettings
+   * @throws IOException
+   */
+  public static TableAdminClient create(
+      InstanceName instanceName, BigtableTableAdminSettings adminSettings) throws IOException {
+    return new TableAdminClient(instanceName, adminSettings);
+  }
+
+  /**
+   * Constructs an instance of TableAdminClient with the given instanceName and
+   * bigtableTableAdminStub
+   *
+   * @param instanceName
+   * @param stub
+   * @throws IOException
+   */
+  public static TableAdminClient create(InstanceName instanceName, BigtableTableAdminStub stub)
+      throws IOException {
+    return new TableAdminClient(instanceName, stub);
+  }
+
+  private TableAdminClient(InstanceName instanceName, BigtableTableAdminSettings adminSettings)
+      throws IOException {
+    this(
+        instanceName,
+        ((BigtableTableAdminStubSettings) adminSettings.getStubSettings()).createStub());
+  }
+
+  private TableAdminClient(InstanceName instanceName, BigtableTableAdminStub stub)
+      throws IOException {
+    Preconditions.checkNotNull(instanceName);
+    Preconditions.checkNotNull(stub);
+    this.instanceName = instanceName;
+    this.stub = stub;
+  }
+
+  /**
+   * Gets the instanceName this client is associated to
+   *
+   * @return InstanceName
+   */
+  public InstanceName getInstanceName() {
+    return instanceName;
+  }
+
+  @Override
+  public void close() {
+    stub.close();
+  }
+
+  /**
+   * Creates a new table with the specified configuration
+   *
+   * <p>Sample code:
+   *
+   * <pre>{@code
+   *  try(TableAdminClient client =  TableAdminClient.create(InstanceName.of("[PROJECT]", "[INSTANCE]"))) {
+   *     CreateTable createTableReq =
+   *         TableAdminRequests.createTable("tableId")
+   *             .addColumnFamily("cf2", GCRULES.maxVersions(10))
+   *             .withGranularity(TimestampGranularity.MILLIS);
+   *
+   *     client.createTable(createTableReq);
+   *   }
+   * }</pre>
+   *
+   * @param createTable
+   * @return Table - the newly created table
+   * @see CreateTable for createTable configurations
+   */
+  public Table createTable(CreateTable createTable) {
+    com.google.bigtable.admin.v2.Table table =
+        this.stub.createTableCallable().call(createTable.toProto(instanceName));
+    return TableAdminResponses.convertTable(table);
+  }
+
+  /**
+   * Creates a new table with the specified configuration asynchronously
+   * <p>Sample code:
+   * <pre>{@code
+   *  try(TableAdminClient client =  TableAdminClient.create(InstanceName.of("[PROJECT]", "[INSTANCE]"))) {
+   *     CreateTable createTableReq =
+   *         TableAdminRequests.createTable("tableId")
+   *             .addColumnFamily("cf2", GCRULES.maxVersions(10))
+   *             .withGranularity(TimestampGranularity.MILLIS);
+   *
+   *     client.createTableAsync(createTableReq);
+   *   }
+   *  }</pre>
```

Comment:
Please fix indent:
`}</pre>` should line up with `<pre>{@code`, the closing bracket should line up with `try`. All indents should be 2 spaces.

Here and everywhere else